### ISSUE TITLE
"No runnable methods" in tests run by biz.aQute.junit

### ISSUE THREAD

**Issue by paulbakker** [bndtools/bnd#501]
It looks like the Junit 4 changes introduced a bug in biz.aQute.junit. 

```
public class MyTest extends TestCase{
    @Test
    public void testList() {
        assertTrue(true);
    }
```

When I run the test the following exception shows:

```
java.lang.Exception: No runnable methods
    at org.junit.runners.BlockJUnit4ClassRunner.validateInstanceMethods(BlockJUnit4ClassRunner.java:169)
    at org.junit.runners.BlockJUnit4ClassRunner.collectInitializationErrors(BlockJUnit4ClassRunner.java:104)
    at org.junit.runners.ParentRunner.validate(ParentRunner.java:355)
    at org.junit.runners.ParentRunner.<init>(ParentRunner.java:76)
    at org.junit.runners.BlockJUnit4ClassRunner.<init>(BlockJUnit4ClassRunner.java:57)
    at org.junit.internal.builders.JUnit4Builder.runnerForClass(JUnit4Builder.java:10)
    at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
    at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
    at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
    at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:26)
    at junit.framework.JUnit4TestAdapter.<init>(JUnit4TestAdapter.java:30)
    at junit.framework.JUnit4TestAdapter.<init>(JUnit4TestAdapter.java:24)
    at aQute.junit.Activator.addTest(Activator.java:413)
    at aQute.junit.Activator.addTest(Activator.java:353)
    at aQute.junit.Activator.createSuite(Activator.java:340)
    at aQute.junit.Activator.test(Activator.java:293)
    at aQute.junit.Activator.run(Activator.java:127)
    at aQute.launcher.Launcher$4.call(Launcher.java:1107)
    at aQute.launcher.Launcher$4.call(Launcher.java:1105)
    at aQute.launcher.Launcher.run(Launcher.java:239)
    at aQute.launcher.Launcher.main(Launcher.java:90)
```

The bnd file contains the correct header: 

```
Test-Cases: ${classes;CONCRETE;EXTENDS;junit.framework.TestCase}
```

When I replace biz.aQute.junit by junit.osgi, everything works as expected (but of course no junit 4 support).

**Comment by kriegfrj** [comment]
This should be fixed by #3145 when it is ready. The new tester implementation will happily run a test class that is a mixture of JUnit 3, 4 and 5.

**Comment by kriegfrj** [comment]
> This should be fixed by #3145 when it is ready. The new tester implementation will happily run a test class that is a mixture of JUnit 3, 4 and 5.

Actually, I discovered that this was not correct. JUnit 5's "Vintage" engine implementation (for running JUnit 3 & 4 tests) behaves the same as the old JUnit 3/4 runner in this regard - you can't mix JUnit 3 & 4 in the same class (if you do, only the JUnit 3 tests will run). The new implementation in #3145 will happily allow you to mix JUnit 3 & 5 or JUnit 4 & 5 in the same class though.

As an alternative solution to this problem of getting feedback on the Eclipse side, with the new architecture in #3145 it would be reasonably easy to have the runner create & register a failed test if it detects a class that extends from `junit.framework.TestCase` and has JUnit 4 annotations in it. That way it will appear as a failed test in Eclipse's JUnit GUI (and in the JUnit XML reports). I might have a look at adding this once #3145 gets merged.
